app/presenters/product_presenter.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class ProductPresenter
+  def initialize(product)
+    @product = product
+  end
+
+  def invalid?(attribute)
+    errors.attribute_names.include? attribute
+  end
+
+  def errors_for(attribute)
+    errors.messages[attribute].map(&:capitalize).join(', ') if invalid?(attribute)
+  end
+
+  def valid_class(attribute)
+    ' is-invalid' if invalid?(attribute)
+  end
```

Comment:
Teraz jak na to patrzę to chyba można kiedys zrobić klasę attribute. Czy to przesada?

Reply:
nope. Schemas są dość popularne produkcyjne https://dry-rb.org/gems/dry-schema/1.5/

Reply:
Spróbowałem użyć tutaj dry-schema ale może zostawię tak jak jest tutaj skoro działa, bo ten nowy commit przynosi więcej zależności i komplikacji
[https://github.com/jakrzus/RubyLoversShop/pull/17/commits/c4f70e6921036e561966a9b324bfc9a5161da2ac](https://github.com/jakrzus/RubyLoversShop/pull/17/commits/c4f70e6921036e561966a9b324bfc9a5161da2ac)